rig/machine_control/machine_controller.py

```diff
@@ -1578,6 +1611,9 @@ def seek(self, n_bytes, from_what=os.SEEK_SET):
             Note that `os.SEEK_END`, `os.SEEK_CUR` and `os.SEEK_SET` are also
             valid arguments.
         """
+        # Flush this write buffer
+        self.flush()
```

Comment:
As mentioned before, I'd prefer seek didn't flush, that way you can always seek before writing without obliterating the buffering behaviour.